spec/functional/bootstrap_download_spec.rb

```diff
@@ -115,6 +117,10 @@ def run_download_scenario
     clean_test_case
 
     winrm_bootstrapper = Chef::Knife::BootstrapWindowsWinrm.new([ "127.0.0.1" ])
+    winrm_bootstrapper.define_singleton_method(:client_builder){nil}
+
+    allow(winrm_bootstrapper.client_builder).to receive(:run)
+    allow(winrm_bootstrapper.client_builder).to receive(:client_path)
```

Comment:
we need a note in here about how this is because Chef 11 doesn't have Chef::Knife::Bootstrap.client_builder.

If we're running against chef 12 (reverting 7aab00da90ad928a5a4cb123b4a300710f7e9005), we can just say:
```
winrm_bootstrapper.client_builder = instance_double("Chef::Knife::Bootstrap::ClientBuilder", :run => nil, :client_path => nil)
```

I think we need to modify travis to test both chef11 + chef12. I'm looking at that.